engine/jobmaster/dm/task_manager.go

```diff
@@ -105,6 +119,10 @@ func (tm *TaskManager) UpdateTaskStatus(taskStatus runtime.TaskStatus) {
 		zap.Uint64("config_modify_revison", taskStatus.CfgModRevision),
 	)
 	tm.tasks.Store(taskStatus.Task, taskStatus)
+	if _, ok := tm.stageGauge[taskStatus.Task]; !ok {
+		tm.stageGauge[taskStatus.Task] = tm.gaugeVec.WithLabelValues(taskStatus.Task)
+	}
+	tm.stageGauge[taskStatus.Task].Set(float64(taskStatus.Stage))
```

Comment:
Will this cause problems with concurrent map reads and writes?

Reply:
After using `withLableValues` directly instead of map, do not exist concurrent problem https://github.com/pingcap/tiflow/pull/7594/commits/3d6be7831b89b453431b905a50cbe8a58cb27022